src/transformers/generation_utils.py

```diff
@@ -189,6 +189,8 @@ def _update_model_kwargs_for_generation(
             model_kwargs["past"] = outputs.past_key_values
         elif "mems" in outputs:
             model_kwargs["past"] = outputs.mems
+        elif "past_buckets_states" in outputs:
+            model_kwargs["past"] = outputs.past_buckets_states
```

Comment:
credit goes to @guillaume-be for catching this bug in https://github.com/huggingface/transformers/pull/8252